bin/add-mod-path.py

```diff
@@ -0,0 +1,159 @@
+```
+ mkdir ~/.module
+ echo 'export MODULEPATH=$HOME/.module:$MODULEPATH'>> ~/.bashrc
+ source ~/.bashrc
+```"""
+
+
+def ask_modpath():
+    try:
+        modpath = os.environ['MODULEPATH']
+    except KeyError:
+        raise RuntimeError(
+            "MODULEPATH not set. Please ensure environment modules are loaded.")
+
+    writeable = []
+    for modulepath in modpath.split(':'):
+        if os.access(modulepath, os.W_OK):
+            writeable.append(modulepath)
+
+    if writeable == []:
+        raise RuntimeError(
+            "Did not find a writeable directory in $MODULEPATH.\nPlease append a writable path.")
+
+    if len(writeable) == 1:
+        return modulepath[0]
+
+    print("Several paths found to store module files. Please select one:")
+    for index, directory in enumerate(writeable):
+        print(f"{index:3d}: {directory}")
+    select = input("Your choice: ")
+    return writeable[int(select)]
+
+
+def ask_name():
+    print("The module will be prefixed with `bout++/`")
+    print("Please enter the name of the module.")
+    name = input("bout++/")
+    return name
+
+
+def ask_bouttop():
+    this = os.path.realpath(__file__)
+    top = "/".join(this.split("/")[:-2])
+    print("Do you want to use `%s` as BOUT_TOP?" % top)
+    while True:
+        inp = input("[Y/n/<alt-path>]")
+        if inp in ["", "y", "Y"]:
+            return top
+        if os.path.isdir(inp):
+            return os.path.realpath(inp)
+        if inp not in "nN":
+            print("`%s` is not a valid directory" % inp)
+
+
+def create_mod(modulepath, name, top):
+    pathlib.Path.mkdir(modulepath+"/bout++", parents=True, exist_ok=True)
+    filename = modulepath+"/bout++/"+name
+    if 'LOADEDMODULES' in os.environ:
+        prereq = []
+        for pr in os.environ['LOADEDMODULES'].split(":"):
+            if pr.startswith("bout++/"):
+                continue
+            prereq.append(f"prereq {pr}\n")
+        prereq = "".join(prereq)
+    else:
+        prereq = ""
+    with open(filename, "w") as f:
+        f.write(f"""#%Module 1.0
+#
+#  BOUT++ module for use with 'environment-modules' package
+#  Created by bout-add-mod-path v0.9
+
+# Only allow one bout++ module to be loaded at a time
+conflict bout++
+# Require all modules that where loaded at generation time
+{prereq}
+
+setenv        BOUT_TOP         {top}
+prepend-path  PATH             {top}/bin
+prepend-path  LD_LIBRARY_PATH  {top}/lib
+prepend-path  PYTHONPATH       {top}/tools/pylib
+prepend-path  IDL_PATH        +{top}/tools/idllib:'<IDL_DEFAULT>'
+""")
+    print(f"created `{filename}`")
+
+
+def writeable_dir(prospective_dir):
+    if not os.path.isdir(prospective_dir):
+        raise Exception(
+            "readable_dir:{0} is not a valid path".format(prospective_dir))
+    if os.access(prospective_dir, os.W_OK):
+        return prospective_dir
+    else:
+        raise Exception(
+            "readable_dir:{0} is not a writeable dir".format(prospective_dir))
```

Comment:
Still needed?

Reply:
Thanks, removed.

I also renamed it, it shouldn't matter for a user that it is a python script :-)